src/EVM.hs

```diff
@@ -1433,12 +1431,12 @@ accessStorage addr slot continue = do
       --     However, without concretization, it may not find things that are actually in the storage
       case readStorage slot c.storage of
         Just x -> case readStorage slotConc c.storage of
-          Just _ -> continue x
-          Nothing -> rpcCall c slotConc
+          Just (Lit _) -> continue x
+          Just _ | not c.external -> continue x
+          _ -> rpcCall c slotConc
```

Comment:
Basically, the issue here was, that the `readStorage` would return a `Just (SLoad...)` in some cases, which is wrong -- it actually didn't find the concrete storage, and the data may not have been retrieved from RPC.